packages/core/src/extension/iframe-resizer-methods.ts

```diff
@@ -0,0 +1,43 @@
+export interface IframeResizerWindow extends Window {
+    parentIFrame: {
+        autoResize: (isEnabled: boolean) => void;
+        setHeightCalculationMethod: (id: string) => void;
+    };
+}
+
+export function isIframeResizerWindow(window: Window): window is IframeResizerWindow {
+    const iframeResizerWindow: IframeResizerWindow = window as IframeResizerWindow;
+
+    return !!iframeResizerWindow.parentIFrame;
+}
+
+export async function enableAutoResizing(taggedElementId?: string): Promise<void> {
+    return new Promise((resolve, reject) => {
+        const waitingForIframeResizer = () => {
+            if (isIframeResizerWindow(window)) {
+                window.parentIFrame.autoResize(true);
+
+                if (taggedElementId) {
+                    const element = document.getElementById(taggedElementId);
+
+                    if (element) {
+                        element.setAttribute('data-iframe-height', '');
+                        window.parentIFrame.setHeightCalculationMethod('taggedElement');
+
+                        resolve();
+                    }
+
+                    reject(new Error(`Element Id not found.`));
+                } else {
+                    window.parentIFrame.setHeightCalculationMethod('bodyOffset');
+
+                    resolve();
+                }
+            } else {
+                setTimeout(waitingForIframeResizer, 100);
```

Comment:
Hi @davidchin , I have updated the PR accrodingly:

- ✅ Modify `ExtensionService#initialize` to return a Promise.
- ✅ Use the success/failed events of `IframeEventPoster` to avoid polling for `window.parentIFrame`.
- ✅ Ensure the Promise is resolved only when the host side receives `iframeResizer`'s initialization message.
- ✅ Add `taggedElementId` and `fixedHeight` to the initialization options.